src/app/webbrowser/morph-browser.qml

```diff
@@ -30,19 +30,19 @@ QtObject {
 
     function init(urls, newSession, incognito) {
         i18n.domain = "morph-browser"
-        if (!newSession && settings.restoreSession && !incognito) {
+        if (!newSession && settings.restoreSession && ! (incognito || settings.incognitoOnStart)) {
             session.restore();
         }
         if (allWindows.length == 0) {
-            windowFactory.createObject(null, {"incognito": incognito}).show();
+            windowFactory.createObject(null, {"incognito": (incognito || settings.incognitoOnStart)}).show();
         }
         var window = allWindows[allWindows.length - 1];
         for (var i in urls) {
             window.addTab(urls[i]).load();
             window.tabsModel.currentIndex = window.tabsModel.count - 1;
         }
         if (window.tabsModel.count === 0) {
-            window.addTab(incognito ? "" : settings.homepage).load();
+            window.addTab(settings.homepage).load();
```

Comment:
Should the homepage be loaded even if the user wants to be in private mode? I'm not sure.